tests/v2/validation/observability/rbac/stackstate_rbac_test.go

```diff
@@ -0,0 +1,257 @@
+package observability
+
+import (
+	"context"
+	"strings"
+
+	"testing"
+
+	"github.com/rancher/rancher/tests/v2/actions/charts"
+	"github.com/rancher/rancher/tests/v2/actions/kubeapi/namespaces"
+	"github.com/rancher/rancher/tests/v2/actions/observability"
+	"github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/rbac"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	extencharts "github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/users"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	uiExtensionsRepo = "https://github.com/rancher/ui-plugin-charts"
+	uiGitBranch      = "main"
+	rancherUIPlugins = "rancher-ui-plugins"
+)
+
+const (
+	project                 = "management.cattle.io.project"
+	rancherPartnerCharts    = "rancher-partner-charts"
+	systemProject           = "System"
+	localCluster            = "local"
+	stackStateConfigFileKey = "stackstateConfigs"
+)
+
+type StackStateRBACTestSuite struct {
+	suite.Suite
+	client                        *rancher.Client
+	session                       *session.Session
+	cluster                       *management.Cluster
+	project                       *management.Project
+	stackstateAgentInstallOptions *charts.InstallOptions
+	stackstateConfigs             observability.StackStateConfigs
+}
+
+func (rb *StackStateRBACTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *StackStateRBACTestSuite) SetupSuite() {
```

Comment:
so overall, I'm seeing:
* create namespace(s)
* add the extensions repo so that the extension can be installed
* install node driver (don't understand this still)
* install CRDs in local
* install extension in local
* install agent on specified cluster

does the order matter? I haven't seen a setup func this long, so just trying to get clarity on everything in here.

Reply:
Yes, the order matters here:

- Extensions are dependent on CRDs.
- Configurations are also dependent on CRDs.
- The agent requires these to be installed before proceeding with the chart installation.

One missing step from the above list is adding configurations in local, which is required after the CRDs are installed.

for the node driver installation, in the UI, it's installed while the CRDs are created, so I'm following that order by installing the node driver first and then continuing with the rest.